lib/ansible/module_utils/common/text/converters.py

```diff
@@ -23,7 +30,33 @@
                                       'surrogate_then_replace'))
 
 
-def to_bytes(obj, encoding='utf-8', errors=None, nonstring='simplerepr'):
+@_t.overload
+def to_bytes(
+    obj: _T,
+    encoding: str = 'utf-8',
+    errors: str | None = None,
+    *,
+    nonstring: _NonStringPassthru,
+) -> _T | bytes: ...
+
+
+@_t.overload
+def to_bytes(
+    obj: object,
+    encoding: str = 'utf-8',
+    errors: str | None = None,
+    *,
+    nonstring: _NonStringOther = 'simplerepr',
+) -> bytes: ...
+
+
+def to_bytes(
+    obj: _T,
+    encoding: str = 'utf-8',
+    errors: str | None = None,
+    *,
+    nonstring: _NonStringAll = 'simplerepr'
+) -> _T | bytes:
```

Comment:
Why is the `nonstring` argument being converted to keyword-only for `to_bytes` and `to_text`? It may be uncommon, but that's a breaking change.

Reply:
Without doing this, it breaks the ability of mypy (maybe others) to properly detect which overload to use based the omission of `nonstring`. It then requires setting `_NonStringPassthru = 'passthru',` which isn't valid, and using `simplerepr` as the default with `_NonStringPassthru` isn't valid.

I did look through core, and all collections and cannot find anyone using `nonstring` without supplying it as a keyword.  I was aware of it being a breaking change, and decided I was ok with that, due to low to no usage that would actually break.

Reply:
And actually setting a default of `passthru` for the one overload doesn't work either.